extensions/amp-fit-text/0.1/test/test-amp-fit-text.js

```diff
@@ -53,6 +60,19 @@ describes.realWin(
         .then(() => ft);
     }
 
+    it('buildDom should have same DOM mutations as buildCallback', () => {
+      const fitText1 = createElementWithAttributes(doc, 'amp-fit-text', {
+        width: '111px',
+        height: '222px',
+      });
+      const fitText2 = fitText1.cloneNode(/* deep */ true);
+
+      new AmpFitText(fitText1).buildCallback();
```

Comment:
For most tests (2) is the more correct thing to do. Since we want to simulate everything the Runtime would do. That makes sense when testing the component's behavior. All the classNames additions etc. are required. In order to use (2), you also need to add the element to the `document` or else there will be errors finding the ampdoc.

For these tests, the goal is to ensure `buildCallback` and `buildDom` result in the same DOM Mutations -- completely isolated from the rest of the mutations that would occur during Runtime bootstrap of an element.

Reply:
As an example, `i-amphtml-notbuilt` is during createdCallback flow of the CE after being added to the DOM. `buildInternal` will then remove that. All of which is completely irrelevant to the test

Reply:
I see, this helps a lot. Thanks for the explanation!

Reply:
I'm really not confident about the "most tests" part. Don't have a great understanding of what we want "unit test" to mean for components (i.e. runtime on or no)